File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    Queue<Integer> q = new LinkedList<Integer>();
    public int solution(int[] numbers, int target) {
        int answer = 0;
        q.offer(0);
        q.offer(0);
        while(q.peek()!=null){
            int level =q.poll();
            int v = q.poll();
            if(level==numbers.length){
                if(v==target)
                    answer++;
            }else {

                q.offer(level + 1);
                q.offer(v + numbers[level]);

                q.offer(level + 1);
                q.offer(v - numbers[level]);
            }
        }
        return answer;
    }   
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let queue = [0]; // 시작점은 0
    for (let num of numbers) {
        let nextQueue = [];
        for (let current of queue) {
            nextQueue.push(current + num); // 더하기 경우
            nextQueue.push(current - num); // 빼기 경우
        }
        queue = nextQueue; // 다음 단계로 이동
    }

    // 최종 결과 중 target과 같은 값의 개수를 세기
    return queue.filter((value) => value === target).length;
}